server/functions/framework/interpreted_function.go

```diff
@@ -100,6 +100,25 @@ func (iFunc InterpretedFunction) VariadicIndex() int {
 	return -1
 }
 
+// We need to call QueryWithBindings. GMS's QueryWithBindings
+// currently wraps the result iterator in a tracking iterator which
+// calls ProcessList.EndQuery() on the ctx which is used to build the
+// query after the iterator is closed.
+//
+// Here we hack to get a subcontext that will not cause the context
+// associated with the top-level query to get canceled when it gets
+// passed to ProcessList.EndQuery.
```

Comment:
Ohh this is interesting! This does not seem like something that was simple to find. This will apply to the rework in GMS as well, so I'll let James know about it for that too.